server/src/main/java/org/elasticsearch/search/profile/query/InternalProfileCollector.java

```diff
@@ -45,17 +44,13 @@ public class InternalProfileCollector implements Collector {
     /**
      * An array of "embedded" children collectors
      */
-    private final InternalProfileCollector[] children;
+    private final InternalProfileCollectorManager[] children;
```

Comment:
this is not ideal, because using the collector forces you to use the corresponding collector manager. I would prefer to find a way to not have this dependency collector -> collector manager. 

Can't we call `newCollector` for all the children in `InternalProfileCollectorManager#newCollector` and provide the children collectors to the collector instead?